src/java/org/apache/cassandra/hints/HintVerbHandler.java

```diff
@@ -89,6 +92,20 @@ else if (!StorageProxy.instance.appliesLocally(hint.mutation))
             // the topology has changed, and we are no longer a replica of the mutation - since we don't know which node(s)
             // it has been handed over to, re-address the hint to all replicas; see CASSANDRA-5902.
             HintsService.instance.writeForAllReplicas(hint);
+
+            HintsService.instance.metrics.incrHintsReceivedForUnownedRanges();
+            if (DatabaseDescriptor.getLogOutOfTokenRangeRequests())
+            {
+                // Log at most 1 message per second
+                NoSpamLogger.log(logger,
+                                 NoSpamLogger.Level.WARN,
+                                 1,
+                                 TimeUnit.SECONDS,
+                                 "Received hint containing mutation from {} for token {} outside valid range",
+                                 message.from(),
+                                 hint.mutation.key().getToken());
```

Comment:
```suggestion
                                 "Receiving hint not owned or pending, example: from {} for a partition token {} in {}.{}",
                                 message.from(),
                                 hint.mutation.key().getToken(),
                                 <keyspace>, <cf>);
```
…?
(does NoSpamLogger provide any indication in the logs that this is but a sample?)
(same suggestion applies to other NoSpamLogger usages)

Reply:
I guess this applies to any `NoSpamLogger` usage across the whole codebase, but no, you would see something like this:

```
WARN  17:14:06 Received mutation from /127.0.0.4:7012 for token -3485513579396041028 outside valid range for keyspace distributed_test_keyspace
```

The interval is pretty small here at 1 second. I would hope operators wouldn't attempt to use this as a comprehensive listing of all out-of-token operations. (Although even if they did, they could cross reference with `outOfRangeTokenWrites`.)

Is this something you'd want to address as a general problem with `NoSpamLogger` or something where our messaging should indicate this is a sample?

Reply:
This would potentially also be something to explain if we document the new options in `cassandra.yaml`, and certainly in the NEWS text.

Reply:
Actually, I take all that back. We should see something like this in the logs:

```
WARN  [MutationStage-1] node1 2024-09-13 12:47:22,264 NoSpamLogger.java:108 - Received mutation from /127.0.0.4:7012 for token -3485513579396041028 outside valid range for keyspace distributed_test_keyspace
```